arcgis-ios-sdk-samples/Analysis/Distance measurement analysis/DistanceMeasurementAnalysisViewController.swift

```diff
@@ -0,0 +1,161 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import UIKit
+import ArcGIS
+
+class DistanceMeasurementAnalysisViewController: UIViewController, AGSGeoViewTouchDelegate {
+    /// The scene displayed in the scene view.
+    let scene: AGSScene
+    /// The location distance measurement analysis.
+    let locationDistanceMeasurement: AGSLocationDistanceMeasurement
+    
+    /// The scene view managed by the view controller.
+    @IBOutlet weak var sceneView: AGSSceneView!
+    /// The view for displaying distances.
+    @IBOutlet weak var distanceView: UIView!
+    @IBOutlet weak var directMeasurementLabel: UILabel!
+    @IBOutlet weak var horizontalMeasurementLabel: UILabel!
+    @IBOutlet weak var verticalMeasurementLabel: UILabel!
+    
+    required init?(coder: NSCoder) {
+        // Create the scene.
+        scene = AGSScene(basemap: .imagery())
+        
+        // Create the surface and set it as the base surface of the scene.
+        let elevationSources = [
+            AGSArcGISTiledElevationSource(url: .worldElevationService),
+            AGSArcGISTiledElevationSource(url: .brestElevationService)
+        ]
+        let surface = AGSSurface()
+        surface.elevationSources.append(contentsOf: elevationSources)
+        scene.baseSurface = surface
+        
+        // Create the building layer and add it to the scene.
+        let buildingsLayer = AGSArcGISSceneLayer(url: .brestBuildingsService)
+        scene.operationalLayers.add(buildingsLayer)
+        
+        // Create the location distance measurement.
+        let startPoint = AGSPoint(x: -4.494677, y: 48.384472, z: 24.772694, spatialReference: .wgs84())
+        let endPoint = AGSPoint(x: -4.495646, y: 48.384377, z: 58.501115, spatialReference: .wgs84())
+        locationDistanceMeasurement = AGSLocationDistanceMeasurement(startLocation: startPoint, endLocation: endPoint)
+        
+        super.init(coder: coder)
+        
+        locationDistanceMeasurement.measurementChangedHandler = { [weak self] _, _, _ in
+            DispatchQueue.main.async {
+                self?.updateMeasurementLabels()
+            }
+        }
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // Configure the scene view.
+        sceneView.scene = scene
+        sceneView.touchDelegate = self
+        let camera = AGSCamera(lookAt: locationDistanceMeasurement.startLocation, distance: 200, heading: 0, pitch: 45, roll: 0)
```

Comment:
One suggestion. In portrait orientation on the phone, focusing on the start point of the analysis leaves half the analysis clipped.

What do you think of this?
``` Swift
let targetGeom = AGSGeometryEngine.union(ofGeometry1: locationDistanceMeasurement.startLocation,
                                         geometry2: locationDistanceMeasurement.endLocation)?.extent.center
                 ?? locationDistanceMeasurement.startLocation
let camera = AGSCamera(lookAt: targetGeom, distance: 200, heading: 0, pitch: 45, roll: 0)
```

Also, I wonder how Qt, Android and .NET have handled this for phone devices in portrait? If they're not considering it, I think this might be a nice option (or at least consideration) to feed back.

Reply:
Actually, this might be a better option (as it only deals with extents and returns an extent):

``` Swift
let targetGeom = AGSGeometryEngine.combineExtents(of: locationDistanceMeasurement.startLocation,
                                         andGeometry: locationDistanceMeasurement.endLocation)?.center
                 ?? locationDistanceMeasurement.startLocation
```